### ISSUE TITLE
Allow arrays as collection for custom pages?

### ISSUE THREAD

**Issue by paulgoetze** [activeadmin/activeadmin#4423]
Would it make sense to handle a collection that is an Array in the [`ActiveAdmin::Helpers::Collection`](https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/helpers/collection.rb)

This would enable you to use an Array instead of a `ActiveRecord::Relation` when creating a table in the content block, e.g.:

``` ruby
ActiveAdmin.register_page 'Page name' do
  content do
    collection = [] # <= this can be an Array of ActiveRecord models, e.g a paginated array

    table_for(collection, class: 'index_table') do
      column :foo
      column :bar
    end

    paginated_collection(collection, entry_name: 'Page name')
  end
end
```

At the moment this throws an `undefined method 'except'` error, because `#except` is called on the collection here: https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/helpers/collection.rb#L7

I guess something like 

``` ruby
def collection_size(c = collection)
  return c.count if c.is_a?(Array) # <= !

  c = c.except :select, :order

  c.group_values.present? ? c.count.count : c.count
end
```

would be sufficient.

**Comment by Fivell** [comment]
@paulgoetze ,  you're welcome to create PR with this change and spec for it

**Comment by paulgoetze** [comment]
@Fivell Wow, it’s been some time. I’m not using ActiveAdmin in any project anymore atm, but I figure it might be useful for some other folks anyway :)

So here, we go with the PR: #4770

I also added the use case from above as a cucumber Scenario.

**Comment by Fivell** [comment]
@paulgoetze , come back and use AA again)

**Comment by Fivell** [comment]
closed via #4770
